pkg/auth/providers/keycloakoidc/keycloak_provider.go

```diff
@@ -107,6 +107,7 @@ func (k *keyCloakOIDCProvider) toPrincipal(principalType string, acct account, t
 		}
 	} else {
 		princ.PrincipalType = GroupType
+		princ.ObjectMeta = metav1.ObjectMeta{Name: k.GetName() + "_" + principalType + "://" + acct.Name}
```

Comment:
`principalType` is used as a flag to determine if you are working with a User principal or a Group principal. Its value is usually either the `UserType`, `GroupType` constant or empty value. This means `principalType` and consts `UserType` or `GroupType` are interchangeable for most situations. Because of the empty value situation, instead of using `principalType` to directly set `princ.PrincipalType`, the constant `UserType` or `GroupType` is used in its place.   

acct.Name will never be empty. For keycloak specificaly, the name is the only required value when creating a group. 

Since we use the ID_TOKEN, to retrieve the name of the group(s) someone belongs to, that name is the only value we have to create the necessary Group principals. All other provider specific logic has to implement Group principals to set the `metav1.ObjectMeta{Name:` using the name value and not an id value to ensure `isAuthorized` can find a matching Group principal if it exists.

Reply:
Sounds great to me. Thanks for the explanation